### PR TITLE
maint: Restructure docs configuration file and improve docs pages

### PR DESCRIPTION
**Pull request by mathbunnyru**

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -1,72 +1,69 @@
...
 
 # -- Options for HTML output -------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
+
+html_theme = "alabaster"
+html_static_path = ["_static"]
+
+# The file above was generated using sphinx 8.1.3 with this command:
+# sphinx-quickstart --project "mamba" --author "QuantStack & mamba contributors" -v "latest" -r "latest" -l en --no-sep --no-makefile --no-batchfile
+# These are custom options for this project
```
Comment (mathbunnyru) [inline docs/source/conf.py:34]: This allows to have really simple configuration file for understanding, where you can easily see, which part of the docs is the standard config, and which is custom for this project.

Updating to new sphinx versions becomes easier and overall we have better distinction between generated code and handwritten.

File: docs/source/conf.py [hunk 2]
```diff
@@ -1,72 +1,69 @@
...
 html_logo = "_static/logo.png"
 html_title = "documentation"
 
 html_theme_options = {
+    "path_to_docs": "docs/source",
+    "repository_branch": "main",
     "repository_url": "https://github.com/mamba-org/mamba",
+    "use_download_button": True,
+    "use_edit_page_button": True,
+    "use_issues_button": True,
```
Comment (mathbunnyru) [inline docs/source/conf.py:46]: I added a few nice options here, which add a few buttons under GitHub logo and also make issues links more beautiful (if there will be any)

File: docs/source/conf.py [hunk 3]
```diff
@@ -1,72 +1,69 @@
...
+
+# Configuration of Breathe Doxygen interopt
+breathe_projects = {"libmamba": os.environ.get("MAMBA_DEV_DOXYGEN_XML_DIR", "../xml")}
+breathe_default_project = "libmamba"
+breathe_default_members = ("members", "undoc-members")
+breathe_show_include = False
+
+# Configure links which should be ignored during linkcheck
+linkcheck_ignore = [
+    r".*github\.com.*#",  # javascript based anchors
```
Comment (mathbunnyru) [inline docs/source/conf.py:68]: GitHub Readme files have javascript based anchors, which causes `linkcheck` errors. I also fixed 2 redirect warnings below.
Now, docs links are correct (and maybe I will add a CI step later)

**Comment by jjerphan** [comment]
I think that it would be appropriate activating the rendering of Pull Requests' documentation rendering first, i.e. what you proposed with https://github.com/mamba-org/mamba/issues/3598.

**Review by JohanMabille — APPROVED** [review]

**Comment by JohanMabille** [comment]
It looks like I don't have the permission to do that, so I would be in favor of merging this PR.

**Review by jjerphan — APPROVED** [review]
Thank you, @mathbunnyru.

**Pull request merged**